src/ol/renderer/canvas/canvastilelayerrenderer.js

```diff
@@ -365,7 +365,7 @@ ol.renderer.canvas.TileLayer.prototype.prepareFrame =
   /** @type {Array.<number>} */
   var zs = Object.keys(tilesToDrawByZ).map(Number);
   zs.sort(ol.array.numberSafeCompareFunction);
-  var opaque = tileSource.getOpaque();
+  var opaque = tileSource.getOpaque(projection);
```

Comment:
So far, most of the tile reprojection handling is in `ol.source.TileImage` so I added it there to keep it in all one place.
The renderer has no special handling related to reprojection and the `opaqueAndNotReprojecting` would make sense only for `ol.source.TileImage` (not for other subclasses of `ol.source.Tile`).

Alternatively, we could rename `getOpaque` to `getOpaqueForProjection` (or keep `getOpaque` unchanged and add `getOpaqueForProjection`) to make it more clear (and consistent with `getTileGridForProjection` and `getTileCacheForProjection` methods).

I have no strong opinion on this, so I'm not opposed to moving the projection equivalence check to the `canvastilelayerrenderer.js` if others also think it would be better. Opinions?